examples/oracle.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""Basic examples of various lifecycle with an OCI instance."""
+
+import logging
+import sys
+from base64 import b64encode
+
+import pycloudlib
+
+cloud_config = """#cloud-config
+runcmd:
+  - echo 'hello' > /home/ubuntu/example.txt
+"""
+
+
+def demo(availability_domain: str, compartment_id: str):
+    """Show example of using the OCI library.
+
+    Connects to OCI and launches released image. Then runs
+    through a number of examples.
+    """
+    with pycloudlib.OCI(
+        "oracle-test",
+        availability_domain=availability_domain,
+        compartment_id=compartment_id,
+    ) as client:
+        with client.launch(
+            image_id=client.released_image("focal"),
+            user_data=b64encode(cloud_config.encode()).decode(),
+        ) as instance:
+            instance.wait()
+            print(instance.instance_data)
+            print(instance.ip)
+            instance.execute("cloud-init status --wait --long")
+            print(instance.execute("cat /home/ubuntu/example.txt"))
+
+            snapshotted_image_id = client.snapshot(instance)
+
+        with client.launch(image_id=snapshotted_image_id) as new_instance:
+            new_instance.wait()
+            new_instance.execute("whoami")
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG)
+    
+    # fill in the  variables here if you'd rather not
+    # pass availability domain and compartment id as arguments via cli
+    AVAILABILITY_DOMAIN = ""
+    COMPARTMENT_ID = ""
+
+    # if no arguments are passed and the variables are not set
```

Comment:
I don't see the purpose of interactive input when a user can simply pass CLI args. It might be more hand-holdy the first time somebody runs a script, but beyond that there are a few downsides.

It's not repeatable. With a CLI args, I can paste the values into the CLI, then recall the same command from my history or throw it in a script. If it's interactive, I need to re-copy and re-paste the same values every single time I run it, and running it within a script becomes much harder.

It can quickly become unmaintainable. Need another arg? What about flags that only apply to specific args? 

What's the harm in seeing a usage line that tells you the correct input?

Reply:
Agreed. I'll nuke the changes i made to the script. thanks for presenting this feedback and thoroughly explaining why. I appreciate it.